metrics/ci_integrations/test/integration/ci/sync_stage/builds/in_progress_builds_sync_stage_test.dart

```diff
@@ -52,8 +110,535 @@ void main() {
+        whenRefreshBuild(inProgressBuilds[0]).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        whenRefreshBuild(inProgressBuilds[1]).thenAnswer(
+          (_) => Future.value(updatedBuilds[1]),
+        );
+
+        await syncStage.call(syncConfig);
+
+        verifyNever(sourceClient.fetchCoverage(argThat(isIn(updatedBuilds))));
+      },
+    );
+
+    test(
+      ".call() does not continue the sync if fetching coverage for a build fails with an error",
+      () async {
+        final inProgressBuild = BuildData(
+          buildNumber: 1,
+          buildStatus: BuildStatus.inProgress,
+          startedAt: startedAtExceedingTimeout,
+        );
+        final inProgressBuilds = [inProgressBuild];
+        final updatedBuilds = inProgressBuilds
+            .map((build) => build.copyWith(buildStatus: BuildStatus.successful))
+            .toList();
+
+        whenFetchInProgressBuilds().thenAnswer(
+          (_) => Future.value(inProgressBuilds),
+        );
+        whenRefreshBuild(inProgressBuild).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        when(
+          sourceClient.fetchCoverage(updatedBuilds[0]),
+        ).thenAnswer((_) => Future.error(error));
+
+        await syncStage.call(syncConfigWithCoverage);
+
+        verifyNever(destinationClient.updateBuilds(destinationProjectId, any));
+      },
+    );
+
+    test(
+      ".call() returns an error if fetching coverage for a build fails with an error",
+      () async {
+        final inProgressBuild = BuildData(
+          buildNumber: 1,
+          buildStatus: BuildStatus.inProgress,
+          startedAt: startedAtExceedingTimeout,
+        );
+        final inProgressBuilds = [inProgressBuild];
+        final updatedBuilds = inProgressBuilds
+            .map((build) => build.copyWith(buildStatus: BuildStatus.successful))
+            .toList();
+
+        whenFetchInProgressBuilds().thenAnswer(
+          (_) => Future.value(inProgressBuilds),
+        );
+        whenRefreshBuild(inProgressBuild).thenAnswer(
+          (_) => Future.value(updatedBuilds[0]),
+        );
+        when(
+          sourceClient.fetchCoverage(updatedBuilds[0]),
+        ).thenAnswer((_) => Future.error(error));
+
+        final result = await syncStage.call(syncConfigWithCoverage);
+
+        expect(result.isError, isTrue);
+      },
+    );
+
+    test(
+      ".call() updates builds with the fetched coverage if the coverage option is enabled in the sync config",
+      () async {
+        final inProgressBuild = BuildData(
+          buildNumber: 1,
+          buildStatus: BuildStatus.inProgress,
+          startedAt: startedAtExceedingTimeout,
+        );
+        final anotherInProgressBuild = BuildData(
+          buildNumber: 2,
+          buildStatus: BuildStatus.inProgress,
+          startedAt: startedAtExceedingTimeout,
+        );
+        final inProgressBuilds = [inProgressBuild, anotherInProgressBuild];
```

Comment:
There is the `inProgressBuilds` variable for the group - should we use it somehow in the last test cases?